### PR TITLE
Type `status` variables as `cudaError_t`

### PR DESCRIPTION
**Pull request by jakirkham**
Possibly related to issue ( https://github.com/rapidsai/cudf/issues/627 )

Since these are C typed variables, make sure they are typed when assigned to avoid casting them to Python objects (at least until that is needed). Will at least delay this to the error handling section of code where we can accept the conversion cost.

### CODE REVIEW

File: python/cudf/cudf/_cuda/gpu.pyx [hunk 1]
```diff
@@ -341,7 +341,7 @@ def deviceGetName(int device):
     """
 
     cdef char* device_name = <char*> malloc(256 * sizeof(char))
-    status = cuDeviceGetName(device_name, 256, device)
+    cdef int status = cuDeviceGetName(device_name, 256, device)
```
Comment (jakirkham) [inline python/cudf/cudf/_cuda/gpu.pyx:344]: Noted we could clean this up a bit as well. Submitted PR ( https://github.com/rapidsai/cudf/pull/4927 ) to do that.

**Comment by kkraus14** [comment]
```
Error compiling Cython file:

------------------------------------------------------------

...



    This function automatically raises CUDARuntimeError with error message

    and status code.

    """

    cdef int version

    cdef cudaError_t status = cudaDriverGetVersion(&version)

                                                 ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:249:50: Cannot assign type 'int' to 'cudaError_t'



Error compiling Cython file:

------------------------------------------------------------

...

    This function automatically raises CUDARuntimeError with error message

    and status code.

    """



    cdef int version

    cdef cudaError_t status = cudaRuntimeGetVersion(&version)

                                                  ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:266:51: Cannot assign type 'int' to 'cudaError_t'



Error compiling Cython file:

------------------------------------------------------------

...

    This function automatically raises CUDARuntimeError with error message

    and status code.

    """



    cdef int count

    cdef cudaError_t status = cudaGetDeviceCount(&count)

                                               ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:282:48: Cannot assign type 'int' to 'cudaError_t'



Error compiling Cython file:

------------------------------------------------------------

...

    This function automatically raises CUDARuntimeError with error message

    and status code.

    """



    cdef int value

    cdef cudaError_t status = cudaDeviceGetAttribute(&value, attr, device)

                                                   ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:304:52: Cannot assign type 'int' to 'cudaError_t'



Error compiling Cython file:

------------------------------------------------------------

...

    This function automatically raises CUDARuntimeError with error message

    and status code.

    """



    cdef cudaDeviceProp prop

    cdef cudaError_t status = cudaGetDeviceProperties(&prop, device)

                                                    ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:324:53: Cannot assign type 'int' to 'cudaError_t'



Error compiling Cython file:

------------------------------------------------------------

...

    This function automatically raises CUDARuntimeError with error message

    and status code.

    """



    cdef char* device_name = <char*> malloc(256 * sizeof(char))

    cdef cudaError_t status = cuDeviceGetName(device_name, 256, device)

                                            ^

------------------------------------------------------------



cudf/_cuda/gpu.pyx:344:45: Cannot assign type 'int' to 'cudaError_t'
```

**Comment by jakirkham** [comment]
Yeah the signatures that were declared here appear to have been wrong.

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART____VERSION.html

**Comment by jakirkham** [comment]
It looks like [`cuDeviceGetName`]( https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1gef75aa30df95446a845f2a7b9fffbb7f ) actually returns [`CUresult`]( https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TYPES.html#group__CUDA__TYPES_1gc6c391505e117393cc2558fff6bfc2e9 ) instead of [`cudaError_t`]( https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1gf599e5b8b829ce7db0f5216928f6ecb6 ). Asking internally to figure out the differences.

**Comment by jakirkham** [comment]
As device error handling was a little bit more involved, have opted to start fresh. Added PR ( https://github.com/rapidsai/cudf/pull/4943 ), which contains much of the same content/ideas here, but handles the device API error handling case in a slightly different way (as was needed).

**Comment by jakirkham** [comment]
As device error handling was a little bit more involved, have opted to start fresh. Added PR ( https://github.com/rapidsai/cudf/pull/4943 ), which contains much of the same content/ideas here, but handles the device API error handling case in a slightly different way (as was needed).

**Pull request closed without merging**